packages/core/src/util/uuidv4.ts

```diff
@@ -1,10 +1,16 @@
-/* istanbul ignore next */
-const crypto = globalThis.crypto || (await import("node:crypto")).webcrypto
-
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export function getCrypto() {
-  return crypto
+  return typeof window === "undefined"
+    ? // eslint-disable-next-line @typescript-eslint/no-var-requires
+      require("node:crypto").webcrypto
+    : window.crypto
```

Comment:
reverted top-level await because it was the only hang up with the cjs builds and it's only used once in hatchify

Reply:
we cannot do dynamic require in esm